### ISSUE TITLE
PTM fine-tuning details

### ISSUE THREAD

**Issue by jesperdlau** [PHOENIXcenter/pi-PrimeNovo#10]
Hi there. Thanks so much for sharing the code and model. Very exciting method!

I am curious about the details of how you fine-tune. You write in the paper: 

> Our training methodology employed a dataset encompassing 21
distinct PTMs, referred to as the 21PTMs dataset, as detailed in ref. 40.
We fine-tuned PrimeNovo for each PTM to ascertain its proficiency in
peptide generation and PTM classification, in accordance with pre-
viously described methods. To ensure dataset balance, we included an
approximately equal number of peptides with and without PTMs,
culminating in a total of 703,606 PSMs for the dataset. The compre-
hensive fine-tuning endeavor across the 21 PTMs allows PrimeNovo to
discern a broad spectrum of PTMs, a capability evidenced by the
exemplary performance metrics for each PTM category depicted in
Fig. 6c.

So to my understanding, you fine-tune on 95% of 2020-Cell-LUAD in a random split. Then for the 21PTM you take the phosphorylation checkpoint and further fine-tune on a subset of each PTM of the 21PTM dataset. Is that correct? If so, how do you split each subset of 21PTM? And do you also fine-tune on the 21PTM phospho subset? 

Thanks again.

**Comment by wyattz23** [comment]
Hi, 2020- Cell-Luad is a separate experiments than 21PTM. They both start from pre-trained PrimeNovo. One of fine-tuned on LUAD data, tested on LUAD test set. One is fine-tuned on each PTM in 21 PTM dataset and tested on corresponding testing part. 

21PTM is split randomly within each PTM. All of our data splitting aims to make sure that there's not peptide overlap between training and testing. We also make sure PTM and no-PTM modification are 1:1 mixed in 21PTM experiments.

**Comment by jesperdlau** [comment]
How large/what is the proportion of each test set of 21PTM?
